xdsl/dialects/snitch_runtime.py

```diff
@@ -1,11 +1,21 @@
 from abc import ABC
-from xdsl.irdl import irdl_op_definition, IRDLOperation, Operand, Operation, SSAValue
+from xdsl.irdl import (
+    irdl_op_definition,
+    IRDLOperation,
+    Operand,
+    Operation,
+    SSAValue,
+    VarOperand,
+)
 from xdsl.ir import OpResult, Dialect
-from xdsl.dialects.builtin import IntegerType, Signedness, IndexType
+from xdsl.dialects.builtin import IntegerType, Signedness, IndexType, i32
+from xdsl.dialects.llvm import LLVMPointerType
 from typing import Annotated
 
 u32 = IntegerType(data=32, signedness=Signedness.UNSIGNED)
 u64 = IntegerType(data=64, signedness=Signedness.UNSIGNED)
+i8 = IntegerType(data=8, signedness=Signedness.SIGNLESS)
+charptr = LLVMPointerType.typed(i8)
```

Comment:
I'm not sure that we want a dependency on llvm for the snitch dialect. Is there any other way to represent this?

Reply:
I completely agree, open to suggestions.
I could for example, let this be emitted as part of the lowering process to the LLVM specific part?

Right now it is implemented like this to make the linking possible as per this
wiki page:
https://github.com/xdslproject/xdsl/wiki/IO-Example-through-LLVM-stdlib

Reply:
@AntonLydike do you have any suggestions?

Reply:
Does anything prevent you from just adding a `SnrtString` type for the moment? Once you get a better handle on what that thing really is, you can still refactor, IMO. Or do you actually go through LLVM IR at some point to lower/execute this?

Reply:
During lowering, you can then lower that to the equivalent of a char* in whatever comes after snrt, i guess.

Reply:
I am actually curious what you want to print? Printf is a C style interface with complex formatting. Do we need such an interface, or would it be sufficient to have something simpler, e.g., just print_int()?